### DISCUSSION TITLE
build failed with an error simple-cons

### DISCUSSION THREAD

**Discussion by AbdulSayyed** [hbstack/.github#85] [Issues and bugs]
Using Ubuntu24.04 with following versions ghafoor@programming-vm:~ [Node: v20.18.0] [Go: go1.23.2] [Hugo: 0.136.2+extended] [SASS: 1.80.3]
npm run dev throws the following error with numbers of SASS warning, tried everything can not resolve. Need some advice. The error is as follows
More info: https://sass-lang.com/d/color-functions
WARN  Dart Sass: file:///home/ghafoor/.cache/hugo_cache/modules/filecache/modules/pkg/mod/github.com/twbs/bootstrap@v5.3.4-0.20241025054014-c2a7d686de56 incompatible/scss/_functions.scss:36:10: red() is deprecated. Suggestion:

color.channel($color, "red", $space: rgb)

More info: https://sass-lang.com/d/color-functions
WARN  Dart Sass: file:///home/ghafoor/.cache/hugo_cache/modules/filecache/modules/pkg/mod/github.com/twbs/bootstrap@v5.3.4-0.20241025054014-c2a7d686de56 incompatible/scss/_functions.scss:36:23: green() is deprecated. Suggestion:

color.channel($color, "green", $space: rgb)

More info: https://sass-lang.com/d/color-functions
WARN  Dart Sass: 317 repetitive deprecation warnings omitted.
Run in verbose mode to see all warnings.
ERROR [icons] simple-icons's color doesn't exist.
WARN  [github.com/hugomods/images] image not found: path/to/your/image.jpg
WARN  [archives] section archives/2016 not found
WARN  [archives] section archives/2016/04 not found
Built in 6305 ms
Error: error building site: logged 1 error(s)
Not worried about other warning, but how to avoid SASS warning and Simple Icon error ? Thanks

**Comment by razonyang** [comment]
Hi, the icons module throws an error by default if the icon is not found, **to ensure that the build result is predictable**.

> ERROR [icons] simple-icons's color doesn't exist.

You can either replace the icon (`color` may doesn't exist or has been deleted) with the existing one.

Or throws a warnging instead by setting the `icons.throw_on_error` site parameter to `false`, and fixed the icon later.

```yaml
// params.yaml
icons:
  throw_on_error: false
```

**Comment by razonyang** [comment]
By the way, the Dart Sass warnings are related to the https://github.com/twbs/bootstrap/issues/40849.

You can pull latest Bootstrap (`hugo mod get github.com/twbs/bootstrap@main`) if that issue was fixed.

**Comment by razonyang** [comment]
FYI, since [github.com/hbstack/hb@v0.16.0](https://github.com/hbstack/hb/releases/tag/v0.16.0) (requires Hugo's v0.139.2 or later), those warnings were ignored by default.
